bot/plotting/random_plot_generator.py

```diff
@@ -95,6 +113,28 @@ def random_plot_generator(
                     "SEI": sei,
                 })
 
+            solvers = [
+                pybamm.CasadiSolver(mode="safe"),
+                pybamm.CasadiSolver(mode="fast with events")
+            ]
+
+            # choose a random solver
+            solver = random.choice(solvers)
+
+            # 0: pre-defined model with a pre-defined chemistry
+            # 1: experiment with summary variable
+            # 2: experiment without summary variables
+            # 3: comparison plots
+            choice = random.randint(0, 3)
+
+            # if testing, don't randomly choose stuff
+            if testing is True and provided_choice is not None:
+                choice = provided_choice
+
+            # Add degradation only if we are plotting summary variables
+            if choice != 1:
+                options = None
+
             models = [
```

Comment:
Very minor point, but you could define 
```python3
            MODELS = [
                pybamm.lithium_ion.DFN(
                    options=options
                ),
                pybamm.lithium_ion.SPM(
                    options=options
                ),
                pybamm.lithium_ion.SPMe(
                    options=options
                ),
            ]
```
etc. at the top of the script. Then if we want to add more models, chemistries etc. you don't need to search the code for the list. Also, you can just see at the top of the script all the different combinations that might be compared. I'm not sure what is recommended style-wise?

Reply:
For this, I have moved every such list outside the function but the `models` list depends upon the plot that we are creating if it a summary variable plot then the models get degradation otherwise not. Should I still move the `models` list out? and then redefine the list when we need to include degradation?

Reply:
Could you do e.g. `MODELS = [pybamm.lithium_ion.SPM, pybamm.lithium_ion.DFN]` then create instances as necessary with options as  `a = models[i]()` and `b = models[i](options={"SEI":"ec reaction limited"})`?

Maybe it's better to leave this as it was since you end up setting different options depending on the plot and the parameter set anyway. Not sure of the best approach. Both are probably ok, so I'll leave it to you :)

Reply:
I tried doing this but it didn't work out, I'll keep the present one for now. Thanks!